digits/job.py

```diff
@@ -349,6 +350,12 @@ def is_persistent(self):
         """
         return self.persistent
 
+    def delete_timeout(self):
```

Comment:
it might be easier to read if you name this `get_delete_timeout()` (and add a `set_delete_timeout()` too for symmetry). Otherwise it may sound as if this function deletes the timeout.